pkg/ccl/sqlproxyccl/tenantdirsvr/test_static_directory_svr.go

```diff
@@ -176,6 +176,67 @@ func (d *TestStaticDirectoryServer) WatchPods(
 	)
 }
 
+// WatchTenants allows callers to monitor for pod update events.
+//
+// WatchTenants implements the tenant.DirectoryServer interface.
+func (d *TestStaticDirectoryServer) WatchTenants(
```

Comment:
comment: I wish we could simplify the sqlproxy -> tenantdir API. In an ideal world there would be two functions:

1. StartTenant(tenant): starts a pod for the tenant if none are ready.
2. WatchTenants() -> ([]Tenant, func(delta Tenant)): list the tenants and wait for changes. The entire tenant is updated any time the crdbtenant changes or one of its pod changes.

I don't think we need to rework this now, but it is something to consider as the API gets more complicated.